### ISSUE TITLE
SOC中损失变为nan的建议和一些疑问

### ISSUE THREAD

**Issue by locusbear** [ZHKKKe/MODNet#143]
`作者您好，我在SOC策略的训练中也出现了loss变为nan的情况，我观察到的情况是前景占比较小的图像会出现这样的问题，而前景占比较大的图像不会出现这样的问题。应该是小占比图像前景在下采样16倍过后可能消失了，这也就导致了
`pred_fg = pred_matte_fg * pred_semantic_fg`
可能全为0

在后面的
`backup_detail_loss = torch.sum(backup_detail_loss, dim=(1,2,3)) / (torch.sum(boundaries, dim=(1,2,3)))`
`backup_matte_loss = torch.sum(backup_matte_loss, dim=(1,2,3)) / (torch.sum(boundaries, dim=(1,2,3)))`
分母为0就会导致nan的出现
我增加了一个1e-10，就恢复正常了
`backup_detail_loss = torch.sum(backup_detail_loss, dim=(1,2,3)) / (torch.sum(boundaries, dim=(1,2,3))+1e-10)`
`backup_matte_loss = torch.sum(backup_matte_loss, dim=(1,2,3)) / (torch.sum(boundaries, dim=(1,2,3))+1e-10)`

我有几个问题想向作者请教：
1 您有对小目标前景图像的优化方案吗，似乎modnet对这样的图像并不友好。
2 我观察到合成的训练数据似乎会造成预测结果边缘呈现锯齿状，您有相应的优化方案吗
3 您有尝试过在modnet中使用其他的损失函数吗，它们的效果和当前的损失相比怎么样
4 我观察到有论文的trimap生成只在mask边缘外面做膨胀，而不是腐蚀膨胀都做，请问您有做过相关的实验吗

非常感谢！

**Comment by ZHKKKe** [comment]
前景色的目的是为了让前景边缘不会出现背景的颜色。融合网络是指一些最近的工作，研究的是通过网络合成前景和背景，使其更加的自然。
